app/src/main/java/org/simple/clinic/forgotpin/createnewpin/ForgotPinCreateNewPinScreenController.kt

```diff
@@ -13,17 +14,36 @@ typealias Ui = ForgotPinCreateNewPinScreen
 typealias UiChange = (Ui) -> Unit
 
 class ForgotPinCreateNewPinScreenController(
-    private val userSession: UserSession
+    private val userSession: UserSession,
+    private val facilityRepository: FacilityRepository
 ) : ObservableTransformer<UiEvent, UiChange> {
 
   override fun apply(upstream: Observable<UiEvent>): ObservableSource<UiChange> {
     val replayedEvents = upstream.compose(ReportAnalyticsEvents()).replay().refCount()
 
-    return replayedEvents
+    return Observable.merge(
+        showUserNameOnScreenCreate(replayedEvents),
+        showFacilityOnScreenCreate(replayedEvents)
+    )
+  }
+
+  private fun showUserNameOnScreenCreate(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
         .ofType<ScreenCreated>()
         .firstOrError()
         .flatMap { userSession.requireLoggedInUser().firstOrError() }
         .map { { ui: Ui -> ui.showUserName(it.fullName) } }
         .toObservable()
   }
+
+  private fun showFacilityOnScreenCreate(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<ScreenCreated>()
+        .firstOrError()
+        .flatMap { userSession.requireLoggedInUser().firstOrError() }
+        .flatMap { facilityRepository.currentFacility(it).firstOrError() }
```

Comment:
Why are you manually turning all observables to singles?

Reply:
Had an offline discussion; decided not to bother too much for manually converting to `Single`s each time.